src/models/main.ts

```diff
@@ -76,4 +80,49 @@ export class CustomMarkdownString extends vscode.MarkdownString {
   ) {
     super(value, supportThemeIcons);
   }
+
+  /**
+   * Construct a tooltip markdown string from a resource and a set of key-value pairs.
+   * @param title The kind of resource (e.g. "Kafka Cluster", "Compute Pool", etc.)
+   * @param iconName Icon to use beside title, if any.
+   * @param keyValuePairs attribute name + from-resource-value pairs. Any undefined value will be skipped.
+   * @param ccloudUrl Optional URL to view this resource in Confluent Cloud.
+   * @returns
+   */
+  static resourceTooltip(
+    title: string,
+    iconName: IconNames | undefined,
+    ccloudUrl: string | undefined,
+    keyValuePairs: KeyValuePairArray,
```

Comment:
This `KeyValuePairArray` seems a bit overkill - any reason we couldn't go with `{ [key: string]: string | undefined }` or `Record<string, string | undefined>`? Should be able to iterate over it like this instead of needing the `.forEach()`:
```ts
    for (const [key, value] of Object.entries(keyValuePairs)) {
      // Skip undefined or empty values
      if (value === undefined || value === "") {
        continue;
      }
      tooltip.appendMarkdown(`\n\n${key}: \`${value}\``);
    }
```